packages/api/src/external/carequality/gateway/index.ts

```diff
@@ -0,0 +1,70 @@
+import { Patient } from "@metriport/core/domain/patient";
+import { out } from "@metriport/core/util/log";
+import { XCPDGateway } from "@metriport/ihe-gateway-sdk";
+import { MetriportError } from "@metriport/shared";
+import { Config } from "../../../shared/config";
+import { getE2eCxIds as getE2eCxIds } from "../../aws/app-config";
+import { getCQDirectoryEntryOrFail } from "../command/cq-directory/get-cq-directory-entry";
+import { getOrganizationsForXCPD } from "../command/cq-directory/get-organizations-for-xcpd";
+import {
+  filterCQOrgsToSearch,
+  searchCQDirectoriesAroundPatientAddresses,
+  toBasicOrgAttributes,
+} from "../command/cq-directory/search-cq-directory";
+import { buildXcpdGateway, cqOrgsToXCPDGateways } from "../organization-conversion";
+
+type Gateways = {
+  v1Gateways: XCPDGateway[];
+  v2Gateways: XCPDGateway[];
+};
+
+export async function gatherXCPDGateways(patient: Patient): Promise<Gateways> {
```

Comment:
This function was moved and is mostly intact - diff below:
- left: original version
- right: new version

<img width="1683" alt="image" src="https://github.com/metriport/metriport/assets/2132564/04d2fce7-8c2f-44d7-826f-7b1d2fd349c5">